File: RepoTransBench/target_projects/Java/Go/tuenti_SmsRadar/public_tests/sms_storage_public_test.go
package public_tests

import (
	"testing"
	"github.com/stretchr/testify/assert"
)

func TestSmsStoragePublic_StoreAndGetSingleSms(t *testing.T) {
	smsStorage := NewSmsStorage()
	sms := NewSms("Charlie", "+1987654321", "Hey there!", 1357924680, INBOX)
	smsStorage.StoreSms(sms)
	retrieved := smsStorage.GetAllSms()
	assert.Equal(t, 1, len(retrieved))
	out := retrieved[0]
	assert.Equal(t, "Charlie", out.GetContact())
	assert.Equal(t, "+1987654321", out.GetAddress())
	assert.Equal(t, "Hey there!", out.GetMessage())
}

func TestSmsStoragePublic_StoreMultipleSms(t *testing.T) {
	smsStorage := NewSmsStorage()
	s1 := NewSms("Delta", "+1234509876", "First msg", 1000000100, SENT)
	s2 := NewSms("Echo", "+1987654322", "Second msg", 1000000200, OUTBOX)
	smsStorage.StoreSms(s1)
	smsStorage.StoreSms(s2)
	list := smsStorage.GetAllSms()
	assert.Equal(t, 2, len(list))
	assert.Equal(t, "First msg", list[0].GetMessage())
	assert.Equal(t, "Second msg", list[1].GetMessage())
}

func TestSmsStoragePublic_StorageIsCleared(t *testing.T) {
	smsStorage := NewSmsStorage()
	smsStorage.StoreSms(NewSms("Foxtrot", "+1324354657", "To clear", 1234000000, DRAFT))
	smsStorage.ClearAll()
	assert.True(t, len(smsStorage.GetAllSms()) == 0)
}

File: RepoTransBench/source_projects/Java/tuenti_SmsRadar/library/src/main/test/com/tuenti/smsradar/SmsStoragePublicTest.java
package com.tuenti.smsradar;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import java.util.List;

public class SmsStoragePublicTest {

    private SmsStorage smsStorage;

    @Before
    public void setUp() {
        smsStorage = new SmsStorage();
    }

    @Test
    public void testStoreAndGetSingleSms() {
        Sms sms = new Sms("Charlie", "+1987654321", "Hey there!", 1357924680L, SmsType.INBOX);
        smsStorage.storeSms(sms);
        List<Sms> retrieved = smsStorage.getAllSms();
        Assert.assertEquals(1, retrieved.size());
        Sms out = retrieved.get(0);
        Assert.assertEquals("Charlie", out.getContact());
        Assert.assertEquals("+1987654321", out.getAddress());
        Assert.assertEquals("Hey there!", out.getMessage());
    }

    @Test
    public void testStoreMultipleSms() {
        Sms s1 = new Sms("Delta", "+1234509876", "First msg", 1000000100L, SmsType.SENT);
        Sms s2 = new Sms("Echo", "+1987654322", "Second msg", 1000000200L, SmsType.OUTBOX);
        smsStorage.storeSms(s1);
        smsStorage.storeSms(s2);
        List<Sms> list = smsStorage.getAllSms();
        Assert.assertEquals(2, list.size());
        Assert.assertEquals("First msg", list.get(0).getMessage());
        Assert.assertEquals("Second msg", list.get(1).getMessage());
    }

    @Test
    public void testStorageIsCleared() {
        smsStorage.storeSms(new Sms("Foxtrot", "+1324354657", "To clear", 1234000000L, SmsType.DRAFT));
        smsStorage.clearAll();
        Assert.assertTrue(smsStorage.getAllSms().isEmpty());
    }
}